tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,344 @@
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel9_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+    },
+}
+
+
+@pytest.fixture
+def function_leapp_cv(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Content View')
+    return module_target_sat.api.ContentView(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture(scope='module')
+def module_leapp_lce(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Lifecycle Environment')
+    return module_target_sat.api.LifecycleEnvironment(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture
+def function_leapp_ak(
+    module_target_sat, function_leapp_cv, module_leapp_lce, module_sca_manifest_org, upgrade_path
+):
+    logger.info(
+        'Creating Leapp Activation Key for RHEL %s -> %s',
+        upgrade_path['source_version'],
+        upgrade_path['target_version'],
+    )
+    return module_target_sat.api.ActivationKey(
+        content_view=function_leapp_cv,
+        environment=module_leapp_lce,
+        organization=module_sca_manifest_org,
+    ).create()
+
+
+@pytest.fixture
+def verify_target_repo_on_satellite(
+    module_target_sat,
+    function_leapp_cv,
+    module_sca_manifest_org,
+    module_leapp_lce,
+    upgrade_path,
+    function_leapp_ak,
+):
+    """Verify target rhel version repositories has enabled on Satellite Server"""
+    target_rhel_major_ver = upgrade_path['target_version'].split('.')[0]
+    cmd_out = module_target_sat.cli.Repository.list(
+        {
+            'search': f'content_label ~ rhel-{target_rhel_major_ver}',
+            'content-view-id': function_leapp_cv.id,
+            'organization-id': module_sca_manifest_org.id,
+            'lifecycle-environment-id': module_leapp_lce.id,
+        }
+    )
+    repo_names = [out['name'] for out in cmd_out]
+    if target_rhel_major_ver == '9':
+        assert RHEL_REPOS['rhel9_bos']['name'] in repo_names
+        assert RHEL_REPOS['rhel9_aps']['name'] in repo_names
+    else:
+        assert RHEL_REPOS['rhel8_bos']['name'] in repo_names
+        assert RHEL_REPOS['rhel8_aps']['name'] in repo_names
+
+
+@pytest.fixture
+def register_host_with_satellite(
+    module_target_sat,
+    custom_leapp_host,
+    module_sca_manifest_org,
+    function_leapp_ak,
+):
+    """Register content host with satellite"""
+    logger.info(
+        'Registering Leapp Host with Satellite - RHEL %s', custom_leapp_host.os_version.major
+    )
+    result = custom_leapp_host.register(
+        module_sca_manifest_org, None, function_leapp_ak.name, module_target_sat
```

Comment:
This one-liner used to be a standard function that @vijaysawant was using instead of fixtures. Later on, we improved the structure of the module by turning the helper functions into fixtures.

This is made as a fixture as the `precondition_check_upgrade_and_install_leapp_tool` fixture assumes that the host is already registered; since those two fixtures are explicitly included as fixtures for the test, it is hard to move the registration to the test itself.

Ideally, the test should only contain steps that are testing the actual Satellite & Leapp integration, and host registration is not related to Leapp.

Reply:
yes, earlier I asked we could move it to either in test itself or in custom_leapp_host fixture, so custom_leapp_host fixture seems like a good place to move.

Reply:
Thanks @vijaysawant